src/GridFS/WritableStream.php

```diff
@@ -173,6 +173,21 @@ public function close(): void
         $this->isClosed = true;
     }
 
+    /**
+     * Delete all files and chunks associated with this stream filename.
+     *
+     * @return int The number of deleted files
+     */
+    public function delete(): int
+    {
+        try {
+            return $this->collectionWrapper->deleteFileAndChunksByFilename($this->file['filename']);
```

Comment:
I realize this is an internal class, so we're not actually introducing a public API here, but a `delete()` method that affects multiple revisions seems odd, which is otherwise an abstraction for writing a single revision.

In the PR description, you said:

> The context resolver mechanism introduced in https://github.com/mongodb/mongo-php-library/pull/1138 doesn't give direct access to the CollectionWrapper which is required for this feature.

If we look at `stream_open`, it accesses a CollectionWrapper after resolving the path to an array of context options. Couldn't the same approach be used here to access a CollectionWrapper directly?

Reply:
I felt deep down that using `ReadableStream` and `WritableStream` wasn't correct. I refactored by adding a private `getContext` method to use `CollectionWrapper` directly for `rename` and `unlink`.